module/zfs/spa.c

```diff
@@ -2318,6 +2332,209 @@ vdev_count_verify_zaps(vdev_t *vd)
 #endif
 
 /*
+ * Determine whether the activity check is required.
+ */
+static boolean_t
+spa_activity_check_required(spa_t *spa, uberblock_t *ub, nvlist_t *config)
+{
+	uint64_t state = 0;
+	uint64_t hostid = 0;
+	uint64_t tryconfig_txg = 0;
+	uint64_t tryconfig_timestamp = 0;
+	nvlist_t *nvinfo;
+
+	if (nvlist_exists(config, ZPOOL_CONFIG_LOAD_INFO)) {
+		nvinfo = fnvlist_lookup_nvlist(config, ZPOOL_CONFIG_LOAD_INFO);
+		(void) nvlist_lookup_uint64(nvinfo, ZPOOL_CONFIG_MMP_TXG,
+		    &tryconfig_txg);
+		(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_TIMESTAMP,
+		    &tryconfig_timestamp);
+	}
+
+	(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_POOL_STATE, &state);
+	(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID, &hostid);
+
+	/*
+	 * Disable the MMP activity check - This is used by zdb which
+	 * is intended to be used on potentially active pools.
+	 */
+	if (spa->spa_import_flags & ZFS_IMPORT_SKIP_MMP)
+		return (B_FALSE);
+
+	/*
+	 * Since the hostid is likely to be different before the pivot than
+	 * it is after the pivot, and since it is hard to see how two nodes
+	 * could safely boot from the same pool anyway, we do not do MMP
```

Comment:
The specific issue we wanted to to avoid  was triggering an activity check during boot because the initramfs sets a different hostid then the rootfs.  This is something which has caused us problems in the past.  But that reasoning was all based on not being able to opt out of MMP via the multihost property.  Now that it's optional we don't really have to worry about that case and I agree it makes sense to do this check even for a root pool.  They did ask for it.